amgut/test/test_auth_handlers.py

```diff
@@ -0,0 +1,133 @@
+from unittest import main
+
+from tornado.escape import url_escape
+
+from amgut.connections import ag_data
+from amgut.test.tornado_test_base import TestHandlerBase
+from amgut import text_locale
+
+
+class TestAuthBasehandler(TestHandlerBase):
+    def test_set_current_user(self):
+        # test if new cookie will be set
+        response = self.get('/auth/register/')
+        self.assertIn('Set-Cookie', response.headers.keys())
+
+        # test that cookie is re-used
+        self.mock_login('tst_ACJUJ')
+        response = self.get('/auth/register/')
+        self.assertNotIn('Set-Cookie', response.headers.keys())
+
+
+class TestAuthRegisterHandoutHandler(TestHandlerBase):
+    def test_get(self):
+        response = self.get('/auth/register/')
+        self.assertEqual(response.code, 200)
+        port = self.get_http_port()
+        self.assertEqual(response.effective_url,
+                         'http://localhost:%d/auth/register/' % port)
+
+    def test_post(self):
+        # test reaction to invalid kit IDs
+        response = self.post('/auth/register/',
+                             {'kit_id': 'wrong_kit_id',
+                              'password': 'test',
+                              'email': 'e@mail',
+                              'email2': 'e@mail',
+                              'participantname': 'joe',
+                              'address': 'street',
+                              'city': 'metropolis',
+                              'state': 'XX',
+                              'zip': '12345',
+                              'country': 'Afghanistan'})
+        port = self.get_http_port()
+        self.assertEqual(
+            response.effective_url,
+            'http://localhost:%d/?loginerror=%s' %
+            (port, url_escape(text_locale['handlers']['INVALID_KITID'])))
+
+        unregistered_kits = ag_data.get_all_handout_kits()
+        if len(unregistered_kits) <= 0:
```

Comment:
I dived into the SQL code. When a kit is registered to an user, the kit is inserted into the table ag_kit and removed from the table ag_handout_kits. During this process, we loose the information in column "created_on". Thus, I cannot revert this function.
To have the DB in the same state as before the test, I would need to write code which anticipates what is going to happen, store the information which is going to be lost somewhere else, perform the SQL operation which I want to test, and create a new function which takes the otherwise stored information to revert the tested SQL operation. This sounds not only quite complicated, but surely introduces potential mistakes + the need to keep code in sync. 
What shall I do??